src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -373,3 +374,28 @@ product-↭ (swap {xs} {ys} x y r) = begin
   (y * x) * product ys ≡⟨ *-assoc y x (product ys) ⟩
   y * (x * product ys) ∎
   where open ≡-Reasoning
+
+------------------------------------------------------------------------
+-- catMaybes
+
+catMaybes-↭ : xs ↭ ys → catMaybes xs ↭ catMaybes ys
+catMaybes-↭ refl = refl
+catMaybes-↭ (trans xs↭ ↭ys) = trans (catMaybes-↭ xs↭) (catMaybes-↭ ↭ys)
+catMaybes-↭ {xs = .(mx ∷ _)} {.(mx ∷ _)} (prep mx xs↭)
+  with mx
+... | nothing = catMaybes-↭ xs↭
+... | just x  = prep x $ catMaybes-↭ xs↭
+catMaybes-↭ {xs = .(mx ∷ my ∷ _)} {.(my ∷ mx ∷ _)} (swap mx my xs↭)
+  with mx | my
```

Comment:
whereas this indicates a missing combinator in `Data.Maybe.Base`, a more general version of `zipWith`.

Reply:
I don't get what you're suggesting here: what does `zipWith` have to do with this proof? 

Given your previous comments, you probably want an eliminator for two maybes simultaneously?
```agda
maybe² : ∀ {A : Set a} {B : Maybe A → Maybe A → Set b}
  → (∀ x y → B (just x) (just y))
  → (∀ x → B (just x) nothing)
  → (∀ y → B nothing (just y))
  → B nothing nothing
  → (x y : Maybe A)
  → B x y
maybe² jj jn nj nn = λ where
  (just x) (just y) → jj x y
  (just x) nothing  → jn x
  nothing  (just y) → nj y
  nothing  nothing  → nn
```
Ofc, the motive will again be necessary:
```agda
catMaybes-↭ {xs = .mx ∷ .my ∷ xs} {.my ∷ .mx ∷ ys} (swap mx my xs↭)  =
    let IH = catMaybes-↭ xs↭ in
    maybe² {B = λ mx my → catMaybes (mx ∷ my ∷ xs) ↭ catMaybes (my ∷ mx ∷ ys)}
           (λ x y → swap x y IH)
           (λ x → prep x IH)
           (λ y → prep y IH)
           IH
           mx my
```

Reply:
I meant exactly this `maybe²` which can legitimately also be called `zipWith` for 2 `Maybe`s.

Having said that, clearly it wouldn't improve things (here). There might be a more-dependent version that would be able to deal with this, but that's out-of-scope for this PR.